spring-integration-sftp/src/test/java/org/springframework/integration/sftp/session/SftpSessionFactoryTests.java

```diff
@@ -343,4 +344,55 @@ protected SftpClient createSftpClient(ClientSession clientSession,
 		}
 	}
 
+	/*
+	 * Verify the socket is closed if authentication fails.
+	 */
+	@Test
+	public void testAuthFailSocketOpen() throws Exception {
+		try (SshServer server = SshServer.setUpDefaultServer()) {
+			server.setPort(0);
+			server.setKeyPairProvider(new SimpleGeneratorHostKeyProvider(new File("hostkey.ser").toPath()));
+			server.start();
+
+			DefaultSftpSessionFactory f = new DefaultSftpSessionFactory();
+			f.setHost("localhost");
+			f.setPort(server.getPort());
+			f.setUser("user");
+			// No authentication provided, will result in authentication failure
+			f.setAllowUnknownKeys(true);
+
+			int n = 0;
+			while (true) {
+				try {
+					f.getSession();
+					fail("Expected Exception");
+				}
+				catch (Exception e) {
+					if (e instanceof IllegalStateException && "failed to create SFTP Session".equals(e.getMessage())) {
+						if (e.getCause() instanceof IllegalStateException) {
+							if (e.getCause().getCause() instanceof ConnectException) {
+								assertThat(n++ < 100).as("Server failed to start in 10 seconds").isTrue();
+								Thread.sleep(100);
+								continue;
+							}
+						}
+					}
+					assertThat(e).isInstanceOf(IllegalStateException.class);
+					assertThat(e.getCause()).isInstanceOf(SshException.class);
+					assertThat(e.getCause().getMessage()).isEqualTo("No more authentication methods available");
+					break;
+				}
+			}
+
+			n = 0;
+			while (n++ < 100 && server.getActiveSessions().size() > 0) {
```

Comment:
In my experimentation it was just like this afater failure:
```
	assertThat(server.getActiveSessions()).hasSize(0);
```
No nee in a loop.
Plus pay attention for more context-specif API from AssertJ.